flask_cors/__init__.py

```diff
@@ -1,546 +1,550 @@
+            setattr(resp, FLASK_CORS_EVALUATED, True)
+
+            return resp
+
+        # If True, intercept OPTIONS requests by modifying the view function,
+        # replicating Flask's default behavior, and wrapping the response with
+        # CORS headers.
+        #
+        # If f.provide_automatic_options is unset or True, Flask's route
+        # decorator (which is actually wraps the function object we return)
+        # intercepts OPTIONS handling, and requests will not have CORS headers
+        if _options.get('automatic_options', True):
+            f.required_methods = getattr(f, 'required_methods', set())
+            f.required_methods.add('OPTIONS')
+            f.provide_automatic_options = False
+
+        return update_wrapper(wrapped_function, f)
+    return decorator
+
+
+class CORS(object):
+    '''
+        Initializes Cross Origin Resource sharing for the application. The
+        arguments are identical to :py:func:`cross_origin`, with the
+        addition of a `resources` parameter. The resources parameter
+        defines a series of regular expressions for resource paths to match
+        and optionally, the associated options
+        to be applied to the particular resource. These options are
+        identical to the arguments to :py:func:`cross_origin`.
+
+        The settings for CORS are determined in the following order:
+            Resource level settings (e.g when passed as a dictionary)
+            Keyword argument settings
+            App level configuration settings (e.g. CORS_*)
+            Default settings
+
+        Note: as it is possible for multiple regular expressions to match a
+        resource path, the regular expressions are first sorted by length,
+        from longest to shortest, in order to attempt to match the most
+        specific regular expression. This allows the definition of a
+        number of specific resource options, with a wildcard fallback
+        for all other resources.
+
+        :param resources: the series of regular expression and (optionally)
+        associated CORS options to be applied to the given resource path.
+
+        If the argument is a dictionary, it is expected to be of the form:
+        regexp : dict_of_options
+
+        If the argument is a list, it is expected to be a list of regular
+        expressions, for which the app-wide configured options are applied.
+
+        If the argument is a string, it is expected to be a regular
+        expression for which the app-wide configured options are applied.
+
+        Default :'*'
+
+        :type resources: dict, iterable or string
+
+    '''
+
+    def __init__(self, app=None, **kwargs):
+        self._options = kwargs
+        if app is not None:
+            self.init_app(app, **kwargs)
+
+    def init_app(self, app, **kwargs):
+        global logger
+        logger = logging.getLogger(app.logger_name + '.cors')
+        logger.addHandler(logging.NullHandler())
```

Comment:
We can't use global here. If there are multiple application or extension instances, this will not behave correctly. 

I think the easiest solution is to just create a helper for getting a logger with either the current application, or else the name "flask.ext.cors".

```Python
try:
    from flask import _app_ctx_stack as stack
except ImportError:
    from flask import _request_ctx_stack as stack

def getLogger():
    if stack.top is not None: # we are in the context of a request
        return logging.getLogger("%s.cors" % current_app.logger_name)
    else:
        return logging.getLogger("flask.ext.cors")
```

Could wrap that up as a static 'logger' property on the CORS class (e.g. CORS.logger with some @property decorator shenanigans.)

I did a quick benchmark to see how expensive calling getLogger is, and it looks quick.

```
Python 2.7.9 (default, Dec 10 2014, 23:46:04)
Type "copyright", "credits" or "license" for more information.

IPython 2.3.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import timeit

In [2]: import logging

In [3]: %timeit logging.getLogger("flask.ext.cors")
100000 loops, best of 3: 4.13 µs per loop

In [4]:
```